### PR TITLE
issue #2469

### LINKED ISSUE #2469: Proper error message and suggestion for AlignTraj on trajectory without mass
**Is your feature request related to a problem? Please describe.**
Coarse-grain trajectory has no assigned mass but it still makes sense to use the AlignTraj. Thus, if one were to run.

```
u = mda.Universe(gro)
protein = u.select_atoms('bynum 1-518')
traj = mda.Universe(xtc, xtc)
AlignTraj(traj, protein, select='bynum 1-518', in_memory=True)
```

He will get

```
  File "pa.py", line 13, in plot
    AlignTraj(traj, protein, select='bynum 1-518', in_memory=True)
  File "/Users//GitHub/mdanalysis/package/MDAnalysis/analysis/align.py", line 668, in __init__
    strict=strict, match_atoms=match_atoms)
  File "/Users//GitHub/mdanalysis/package/MDAnalysis/analysis/align.py", line 1187, in get_matching_atoms
    mass_mismatches = (np.absolute(ag1.masses - ag2.masses) > tol_mass)
  File "/Users//GitHub/mdanalysis/package/MDAnalysis/core/groups.py", line 2294, in __getattr__
    cls=self.__class__.__name__, attr=attr))
AttributeError: AtomGroup has no attribute masses
```

I felt that there should be certain options to disable the check on mass if the weight is not set to mass.

Say if one run 

`AlignTraj(traj, protein, select='bynum 1-518', in_memory=True, weights=None)`

Then the AlignTraj should work without checking for mass.

### PR DESCRIPTION
**Pull request by siddharthjain1611**
Fixes #2469 

Changes made in this Pull Request:
 keeps match_atoms=True
get_matching_atoms was looking for 'mismatching' atoms by checking if anyone pair has a difference higher than tol_mass. This change keeps match_atoms=True.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/align.py [hunk 1]
```diff
@@ -1354,8 +1354,8 @@ def get_atoms_byres(g, match_mask=np.logical_not(mismatch_mask)):
                           "Try to improve your selections for mobile and reference.")
                 logger.error(errmsg)
                 raise SelectionError(errmsg)
-                
-    if match_atoms:
+
+    if match_atoms and hasattr(ag1, 'masses') and hasattr(ag2, 'masses'):
```
Comment (siddharthjain1611) [inline package/MDAnalysis/analysis/align.py]: Line where the code was changed

**Comment by xiki-tempula** [comment]
I felt that instead of just silently skipping the match_atom test.
Adding another except AttributeError and then issue a warning saying that mass match has been skipped due to no mass would be better.
Besides, you probably need to fix the test.

**Comment by orbeckst** [comment]
@xiki-tempula thank you for looking over the PR, your comments are very helpful. 

For reviewing gsoc-related PRs it would be super-helpful if you did a code *review* (instead of just posting a comment) and then select "request changes" to be extra clear what is expected of students. You can always ping another core dev for a second opinion and one of the @MDAnalysis/gsoc-mentors / core devs will have to sign off on the PR anyway.

File: package/MDAnalysis/analysis/align.py (hunk 1 above)
Comment (xiki-tempula) [reply package/MDAnalysis/analysis/align.py]: It might be better if this line stays the same and a warning is issued when no mass can be found (potentially via another except).
The format will be similar to `warnings.warn(msg, category=SelectionWarning)`.
Please read the doc to check the types of warnings available and think of a meaningful message.

**Comment by siddharthjain1611** [comment]
Thank you so much @xiki-tempula sir and @orbeckst sir for your suggestions, Please allow me to work on this and improve this further.

**Comment by orbeckst** [comment]
What was the reason for closing the PR?

**Comment by siddharthjain1611** [comment]
sorry sir, but I thought that I had done something terribly wrong, therefore in order to correct my mistake with s new pr I closed this.
Anyways, sir, it would be a great help if you could help me out with the resources which can help me out with testing the changes locally.

**Comment by orbeckst** [comment]
What is the number of the new PR?

**Comment by siddharthjain1611** [comment]
I've not created one though but I suppose creating an `Attribute Error except` would solve the problem. Please correct if this is correct. I would love your suggestions as I'm a beginner to Molecular Dynamics in Python.

**Comment by xiki-tempula** [comment]
@siddharthjain1611 The idea seems sound to me.
I think the expected behaviour is that when there is no mass. The program executes but issue a warning saying that the mass check has been skipped instead of an error.
The rationale is that for some systems, the mass is not present. However, the program should still run as there is no reason why it shouldn't. However, since the mass check has been skipped, the user need to be warned of the possibility that the atoms are not matched as they thought.

**Comment by RMeli** [comment]
@siddharthjain1611 

The [User Guide](https://www.mdanalysis.org/UserGuide) contains a lot of information about [setting up a developer environment](https://www.mdanalysis.org/UserGuide/contributing_code.html#creating-a-development-environment),  [running tests locally](https://www.mdanalysis.org/UserGuide/testing.html#running-the-test-suite) and [writing new tests](https://www.mdanalysis.org/UserGuide/testing.html#writing-new-tests).

In short, you will need to install `pytest` to run the tests in `testsuite/MDAnalysisTests`.

**Comment by siddharthjain1611** [comment]
`except AttributeError:`
 `errmsg=("Failed to check the mass matches")`
`raise_from(SelectionError(errmsg), None)`
I think that above code should fix the issue.

**Comment by siddharthjain1611** [comment]
I've created a PR #2614. Please suggest if any changes required.
